langlib/lang.array/src/main/ballerina/array.bal

```diff
@@ -166,7 +166,7 @@ public isolated function setLength((any|error)[] arr, int length) returns () = @
 # + val - member to search for
 # + startIndex - index to start the search from
 # + return - index of the member if found, else `()`
-public isolated function indexOf(PureType[] arr, PureType val, int startIndex = 0) returns int? = @java:Method {
+public isolated function indexOf(AnydataType[] arr, AnydataType val, int startIndex = 0) returns int? = @java:Method {
```

Comment:
Do we have negative tests for these already? With non-pure-typed values?

Reply:
No do we need to add negative test cases for this?  I thought it is something out of scope.

Reply:
Since this particular type param is used only in these two functions, was wondering if we should have a simple negative test?